dashboards-observability/public/components/common/search/search.tsx

```diff
@@ -83,20 +85,22 @@ export const Search = (props: any) => {
     flyout = <PPLReferenceFlyout module="explorer" closeFlyout={closeFlyout} />;
   }
 
-  const saveButton = (
-    <EuiButton
-      iconSide="right"
-      onClick={() => {
-        setIsSavePanelOpen((staleState) => {
-          return !staleState;
-        });
-      }}
-      data-test-subj="eventExplorer__saveManagementPopover"
-      iconType="arrowDown"
-    >
-      Save
-    </EuiButton>
-  );
+  const Savebutton = useMemo(() => {
```

Comment:
Why do we need to useMemo here?? It doesn't seem like `searchBarConfigs` or `selectedSubTabId` affect the save button?

Reply:
yea, previously was used to toggle disable flag...missed this